uqcsbot/api.py

```diff
@@ -1,15 +1,12 @@
-from typing import List
+from typing import List, Iterable, AsyncIterable, AsyncGenerator, Generator, Any
+from functools import partial
 from slackclient import SlackClient
+import asyncio
 
 
 # see https://api.slack.com/docs/pagination for details
-def paginate_api_call(client: SlackClient, method: str, *args, **kwargs) -> List[dict]:
-    pages = []
-    while kwargs.get("cursor") != "":
-        page = client.api_call(method, *args, **kwargs)
-        pages.append(page)
-        kwargs["cursor"] = page["response_metadata"]["next_cursor"]
-    return pages
+def paginate_api_call(client: SlackClient, method: str, **kwargs) -> List[dict]:
```

Comment:
I don't think we need this method any more, I believe we can replace the usage in `get_members` with `api.conversations.members.paginate(channel=self.id)`

This change doesn't block merging in though, in case you wanted to leave it for later.